src/plugins/pluginHooks.ts

```diff
@@ -0,0 +1,17 @@
+import {useCallback, useContext, useMemo} from 'react';
+
+import PluginsContext from './PluginsContext';
+import {orderArray} from './utils';
+
+export const usePluginState = <T>(name: string) => {
+  const {scope} = useContext(PluginsContext);
+  const setStateCallback = useCallback((value: any) => scope.setState.bind(name, value), [name, scope]);
```

Comment:
It should be either:
```suggestion
  const setStateCallback = useMemo(() => scope.setState.bind(null, name), [name, scope]);
```
Or:
```suggestion
  const setStateCallback = useCallback(scope.setState.bind(null, name), [name, scope]);
```